src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs

```diff
@@ -259,7 +260,14 @@ private static void ProcessResourceFile(string fileName, Stream input, string[]
 					throw new InvalidOperationException($"The file {fileName} is not a resource file");
 				}
 
+				bool adjustKeyTransformationForV2 = false;
 				var version = reader.ReadInt32();
+				if(version == 2)
+				{
+					version = UPRIVersion;
+					adjustKeyTransformationForV2 = true;
+				}
+
 				if (version != UPRIVersion)
 				{
 					throw new InvalidOperationException($"The resource file {fileName} has an invalid version (got {version}, expecting {UPRIVersion})");
```

Comment:
Why adding code to support v2 if you're immediately throwing an unsupported version right after?